### ISSUE TITLE
Low Latency Streaming: Threshold for catchup mechanism

### ISSUE THREAD

**Issue by dsilhavy** [Dash-Industry-Forum/Live#70]
As described here Dash-Industry-Forum/dash.js#3300 and here Dash-Industry-Forum/dash.js#3316

Right now it is possible to define a target delay and playback rate catchup parameters in the ServiceDescription element of the MPD. However, if the user seeks within the DVR window it might be reasonable to ignore the increase in live delay and continue playing with a playback rate of 1 at the desired time.

We addressed this issue by adding a new settings parameter to dash.js which can be used as a threshold. For instance, if this parameter is set to 8, the catchup mechanism is only applied if currentLiveLatency < 8.

Would this be something worth adding to the ServiceDescription element as well?

**Comment by poolec** [comment]
It seems to me it's more about what the viewer intends than the particular delay behind the live edge.  That translates into what actions the viewer has performed.

For example, if poor network conditions result in you slipping back beyond 8 seconds, giving up trying to maintain the target latency seems the wrong thing to do.

On the other hand, if the viewer has themselves skipped back a bit, it may be reasonable to assume they're happy with that and no longer want to keep at the live edge (unless they seek forward again).

**Comment by dsilhavy** [comment]
I agree on these two use cases, but I think if the player is slipping back beyond the threshold for network reasons it should seek to the target latency instead of trying to adjust the playback rate. 

In any case I think it is reasonable to introduce a threshold parameter for the catchup mechanism.

**Comment by haudiobe** [comment]
Create 2 states: 
* playback at controlled latency
   * at live egde: use service description
   * if user chooses playback position and latency, the overwrite the TargetLatency, but keep the rest (min max, adjusted as well)
* playback without latency control, rebuffering may change latency.

state changes and triggers:
* play => use service description at live egde, if seek to live edge same is happening again
* seek => initiate the no-latency control mode
* setTargetLatency and play would operate at live with an app defined target latency
